src/aae_keystore.erl

```diff
@@ -191,48 +222,84 @@
     % needs to be calculated (like IndexN in native stores)
 
 
--export_type([parallel_stores/0,
-                native_stores/0,
-                manifest/0,
-                objectspec/0,
-                value_element/0,
-                range_limiter/0,
-                segment_limiter/0,
-                modified_limiter/0,
-                count_limiter/0,
-                rebuild_prompts/0,
-                bucket/0,
-                key/0]).
+-export_type(
+    [
+        leveled_options/0,
+        parallel_stores/0,
+        native_stores/0,
+        manifest/0,
+        objectspec/0,
+        value_element/0,
+        range_limiter/0,
+        segment_limiter/0,
+        modified_limiter/0,
+        count_limiter/0,
+        rebuild_prompts/0,
+        bucket/0,
+        key/0
+    ]
+).
 
 
 %%%============================================================================
 %%% API
 %%%============================================================================
 
+-spec store_generate_backendoptions() -> leveled_options().
+store_generate_backendoptions() ->
+    ?LEVELED_DEFAULTS.
+
+-spec store_setbackendoption(
```

Comment:
You seem to do some runtime type checking here and ignore values that are out of bound.
This is needed, because these values come from the outside and you cannot detect that they are out-of-bound by typechecking.

Would it be a good idea to turn the logic around. Check whether it is out of bound, if so write a log message and drop the provided value by reverting to the default?

Reply:
The intention here is to enable riak_kv to pass the right configuration type in a startup.  From the operator perspective, the configuration would be passed into riak_kv via cuttlefish configuration, which should type-check and report an error to the user.

So this is safeguarding against a mistake in development, rather than in operation